src/io.jl

```diff
@@ -114,7 +148,14 @@ function load(fpath::String; gamma::Union{Nothing,Float64}=nothing)
     png_read_end(png_ptr, info_ptr)
     png_destroy_read_struct(Ref{Ptr{Cvoid}}(png_ptr), Ref{Ptr{Cvoid}}(info_ptr), C_NULL)
     close_png(fp)
-    return transpose(buffer)
+    buffer = permutedims(buffer, (2, 1))
+    if expand_paletted || color_type != PNG_COLOR_TYPE_PALETTE
+        return buffer
+    else
+        # We got 0-based indices back from libpng and converting to 1-based could overflow UInt8.
+        # Using UInt16 for index would cost us large part of the savings provided by IndirectArray.
+        return IndirectArray(buffer, OffsetArray(palette, 0:length(palette)-1))
```

Comment:
I think this is equivalent to `palette[buffer .+ 1]` but not sure if they have the same performance.

Reply:
Ah, that is more elegant, thanks!